File: stitch_multifaceted_educator_brand_website/website/update_theme.js
const fs = require('fs');
const path = require('path');

const dir = '.';
const newColors = `                    colors: {
                        primary: '#FF7A00', // Bright Cyan
                        secondary: '#FF7A00', // Bright Cyan
                        background: '#0a0a0a', // Deep Dark Blue/Teal
                        surface: '#171717', // Slightly lighter Dark Blue
                        'on-surface': '#ffffff',
                        'on-surface-variant': '#9ca3af'
                    }`;

const regex = /colors:\s*\{[^\}]+\}/;

let count = 0;
fs.readdirSync(dir).forEach(file => {
    if (file.endsWith('.html')) {
        const filePath = path.join(dir, file);
        let content = fs.readFileSync(filePath, 'utf-8');
        let newContent = content.replace(regex, newColors);
        if (newContent !== content) {
            fs.writeFileSync(filePath, newContent, 'utf-8');
            count++;
        }
    }
});
console.log(`Updated ${count} HTML files in root.`);

const componentsDir = './components';
if (fs.existsSync(componentsDir)) {
    fs.readdirSync(componentsDir).forEach(file => {
        if (file.endsWith('.html')) {
            const filePath = path.join(componentsDir, file);
            let content = fs.readFileSync(filePath, 'utf-8');
            let newContent = content.replace(regex, newColors);
            if (newContent !== content) {
                fs.writeFileSync(filePath, newContent, 'utf-8');
                console.log(`Updated ${file} in components.`);
            }
        }
    });
}

File: stitch_multifaceted_educator_brand_website/website/update_theme.py
import os
import re

directory = '.'
new_colors = """                    colors: {
                        primary: '#54D5F2', // Bright Cyan
                        secondary: '#54D5F2', // Bright Cyan
                        background: '#041520', // Deep Dark Blue/Teal
                        surface: '#0b2336', // Slightly lighter Dark Blue
                        'on-surface': '#ffffff',
                        'on-surface-variant': '#a0c4d8'
                    }"""

count = 0
for filename in os.listdir(directory):
    if filename.endswith(".html"):
        filepath = os.path.join(directory, filename)
        with open(filepath, 'r', encoding='utf-8') as file:
            content = file.read()
        
        new_content = re.sub(r'colors:\s*\{[^\}]+\}', new_colors, content, flags=re.DOTALL)
        
        if new_content != content:
            with open(filepath, 'w', encoding='utf-8') as file:
                file.write(new_content)
            count += 1

print(f"Updated {count} HTML files.")
